### PR TITLE
chore: add checkout step to each job in trigger kong image test workflow

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:

Add `checkout repository` step in each job of trigger Kong image validate workflow.

**Which issue this PR fixes**:

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: .github/workflows/validate_kong_image_trigger_via_label.yaml [hunk 1]
```diff
@@ -14,6 +14,9 @@ jobs:
       ISSUE_NUMBER: ${{ github.event.issue.number }}
       GH_TOKEN: ${{ secrets.K8S_TEAM_BOT_GH_PAT }}
     steps:
+      - name: checkout repository
+        id: checkout_kic_repo
```
Comment (pmalek) [inline .github/workflows/validate_kong_image_trigger_via_label.yaml:18]: I don't see `checkout_kic_repo` being used anywhere. Can we remove this from this file?

Comment (randmonkey) [reply .github/workflows/validate_kong_image_trigger_via_label.yaml]: Removed this part in all jobs.

File: .github/workflows/validate_kong_image_trigger_via_label.yaml [hunk 2]
```diff
@@ -14,6 +14,8 @@ jobs:
       ISSUE_NUMBER: ${{ github.event.issue.number }}
       GH_TOKEN: ${{ secrets.K8S_TEAM_BOT_GH_PAT }}
     steps:
+      - name: checkout repository
+        uses: actions/checkout@v4
```
Comment (tao12345666333) [inline .github/workflows/validate_kong_image_trigger_via_label.yaml:18]: I think this job doesn't need to checkout the repo, right?

File: .github/workflows/validate_kong_image_trigger_via_label.yaml [hunk 3]
```diff
@@ -30,6 +32,8 @@ jobs:
       kong_image_major_version: ${{ steps.semver_parser.outputs.major }}
       kong_image_minor_version: ${{ steps.semver_parser.outputs.minor }}
     steps:
+     - name: checkout repository
+       uses: actions/checkout@v4
```
Comment (tao12345666333) [inline .github/workflows/validate_kong_image_trigger_via_label.yaml:36]: ditto

File: .github/workflows/validate_kong_image_trigger_via_label.yaml (hunk 2 above)
Comment (randmonkey) [reply .github/workflows/validate_kong_image_trigger_via_label.yaml:18]: This requires checkout of repo. If we do not checkout, the step will fail with the error 
```
failed to run git: fatal: not a git repository (or any of the parent directories): .git
```
.

File: .github/workflows/validate_kong_image_trigger_via_label.yaml (hunk 3 above)
Comment (randmonkey) [reply .github/workflows/validate_kong_image_trigger_via_label.yaml]: Deleted. This job does not run `gh` command and does not refer to paths in the repo, so I guess it is OK not to checkout here.

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**